### ISSUE TITLE
Probleme beim update von raspberrypi-kernel (1.20200902-1) 

### ISSUE THREAD

**Issue by voreli** [alexreinert/piVCCU#300]
Hi, wollte gerade den raspberry updaten. Bei raspberrypi-kernel (1.20200902-1) geht es nicht weiter. 
Leider weiß ich nicht weiter.

root@armbianbuild:~# dpkg -C
The following packages have been unpacked but not yet configured.
They must be configured using dpkg --configure or the configure
menu option in dselect for them to work:
 libraspberrypi-bin   Miscellaneous Raspberry Pi utilities
 libraspberrypi-dev   EGL/GLES/OpenVG/etc. libraries for the Raspberry Pi's Vid
 libraspberrypi-doc   EGL/GLES/OpenVG/etc. libraries for the Raspberry Pi's Vid
 libraspberrypi0      EGL/GLES/OpenVG/etc. libraries for the Raspberry Pi's Vid
 rpi-eeprom           Raspberry Pi 4 boot EEPROM updater

The following packages are only half configured, probably due to problems
configuring them the first time.  The configuration should be retried using
dpkg --configure <package> or the configure menu option in dselect:
 raspberrypi-bootloader Raspberry Pi bootloader
 raspberrypi-kernel   Raspberry Pi bootloader

root@armbianbuild:~# apt-get -y upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
7 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_GB.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_GB.UTF-8").
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
Setting up raspberrypi-kernel (1.20200902-1) ...
Removing 'diversion of /boot/kernel.img to /usr/share/rpikernelhack/kernel.img by rpikernelhack'
dpkg-divert: error: unable to change ownership of target file '/boot/kernel.img.dpkg-divert.tmp': Operation not permitted
dpkg: error processing package raspberrypi-kernel (--configure):
 installed raspberrypi-kernel package post-installation script subprocess returned error exit status 2
Setting up raspberrypi-bootloader (1.20200902-1) ...
Removing 'diversion of /boot/start.elf to /usr/share/rpikernelhack/start.elf by rpikernelhack'
dpkg-divert: error: unable to change ownership of target file '/boot/start.elf.dpkg-divert.tmp': Operation not permitted
dpkg: error processing package raspberrypi-bootloader (--configure):
 installed raspberrypi-bootloader package post-installation script subprocess returned error exit status 2
dpkg: dependency problems prevent configuration of libraspberrypi0:
 libraspberrypi0 depends on raspberrypi-bootloader (= 1.20200902-1); however:
  Package raspberrypi-bootloader is not configured yet.

dpkg: error processing package libraspberrypi0 (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of rpi-eeprom:
 rpi-eeprom depends on raspberrypi-bootloader (>= 1.20190819); however:
  Package raspberrypi-bootloader is not configured yet.

dpkg: error processing package rpi-eeprom (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libraspberrypi-doc:
 libraspberrypi-doc depends on libraspberrypi0 (= 1.20200902-1); however:
  Package libraspberrypi0 is not configured yet.

dpkg: error processing package libraspberrypi-doc (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libraspberrypi-bin:
 libraspberrypi-bin depends on libraspberrypi0 (= 1.20200902-1); however:
  Package libraspberrypi0 is not configured yet.

dpkg: error processing package libraspberrypi-bin (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libraspberrypi-dev:
 libraspberrypi-dev depends on libraspberrypi0 (= 1.20200902-1); however:
  Package libraspberrypi0 is not configured yet.

dpkg: error processing package libraspberrypi-dev (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 raspberrypi-kernel
 raspberrypi-bootloader
 libraspberrypi0
 rpi-eeprom
 libraspberrypi-doc
 libraspberrypi-bin
 libraspberrypi-dev
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@armbianbuild:~#

**Comment by alexreinert** [comment]
Das ist zwar kein piVCCU Probem, sondern ein Raspbian Problem, aber dennoch:

Ist das Verzeichnis /usr/share/rpikernelhack/ vorhanden und hat es Schreibrechte?

**Comment by voreli** [comment]
Hi, ja vorhanden und

drwxr-xr-x   3 root 1root  4096 Oct 14 20:28 rpikernelhack

**Comment by alexreinert** [comment]
Das hier würde passen: https://www.raspberrypi.org/forums/viewtopic.php?t=253259
TL;DR; Wie groß ist die /boot Partition und wieviel Platz ist da noch?

**Comment by voreli** [comment]
Danke, den thread hatte ich auch schon gesehen. Das ist es leider nicht.

root@armbianbuild:/usr/share# df

Filesystem     1K-blocks    Used Available Use% Mounted on

/dev/root       30372508 3542192  25558068  13% /

devtmpfs         1827808       0   1827808   0% /dev

tmpfs            1959904       0   1959904   0% /dev/shm

tmpfs            1959904   25008   1934896   2% /run

tmpfs               5120       4      5116   1% /run/lock

tmpfs            1959904       0   1959904   0% /sys/fs/cgroup

/dev/mmcblk0p1    258095   17403    240693   7% /boot

tmpfs             391980       0    391980   0% /run/user/1002

root@armbianbuild:/usr/share#

**Comment by juju1337** [comment]
Ich hab ein ähnliches Problem beim Aufsetzen auf einem frischem Raspi OS mit der o.g. Kernel-Version.

apt install pivccu3 bricht mit der dpkg Fehlermeldung ab.

**Comment by alexreinert** [comment]
Dann mach bitte einen neuen Issue mit voller Fehlermeldung auf.
